Automattic/wp-calypso
# Site Migration: Enable to manage when the site migration key will be loaded

Part of #89760

## Proposed Changes
It is part of #89760 where we need to control a sequence of requests to ensure a site is ready to run the migrate-guru migration. 

I am going to use this feature in further PRs. 


*  Introduce the `enable` prop on the use-site-migration-key hook to manage when we want to start fetching the key. 
* Fix the typo on the file name

## Testing Instructions
* Automated tests cover the change https://github.com/Automattic/wp-calypso/pull/90086/files#diff-775355931667ac238518fab522e40789af941e66df844fc570fc0f2f5bd2a918R39-R51
* The current site migration instructions is the only module using this hook, and it is also covered by automated tests as you can see [here](https://github.com/Automattic/wp-calypso/blob/98a83c097541eb4daffcc77f3a653d9316ec58b8/client/landing/stepper/declarative-flow/internals/steps-repository/site-migration-instructions/test/index.tsx?plain=1#L21).
